.github/workflows/bump-dependencies.yml

```diff
@@ -0,0 +1,179 @@
+            get_upstream_version: |
+              curl -s -o /dev/null --location --range 0-5 --write-out '%{url_effective}' https://sourceforge.net/projects/nsis/files/latest/download |
+                grep -oP '.*/nsis-\K[0-9.]+(?=-setup\.)'
+            # This pattern is a bit special as it has to match twice in a single line.
+            # Therefore, we have to be very careful to avoid consuming too much pattern space.
+            # This is why a positive lookahead is used instead of direct matching:
+            local_version_regex: (.*"nsis-|.*\/NSIS.20.\/|\/nsis-)([0-9.]+)(".*|(?=\/nsis-)|\.zip.*)
+
+          - name: ASIO-SDK
+            changelog_name: ASIO SDK (Windows-only)
+            get_upstream_version: |
+              curl -s -o /dev/null --location --range 0-5 --write-out '%{url_effective}' https://www.steinberg.net/asiosdk |
+                grep -oP '.*asiosdk_\K.*(?=\.zip)'
+            local_version_regex: (.*["\/]asiosdk_)([^"]+?)(".*|\.zip.*)
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ssh-key: ${{ secrets.BUMP_DEPENDENCIES_SSH_DEPLOY_KEY || 'fail-due-to-missing-ssh-key-as-secret' }}
+          fetch-depth: '0'  # we create/compare new branches and therefore require full history
+      - env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          set -eu
+          files=( .github/{autobuild,workflows}/* windows/*.ps1 )
+          upstream_version="$(${{ matrix.components.get_upstream_version }})"
+          local_version="$(perl -nle 'print "$2" if /${{ matrix.components.local_version_regex }}/i' "${files[@]}" | sort --reverse --version-sort | head -n1)"
+          if [[ -z "$upstream_version" ]]; then
+            echo "failed to extract upstream version"
+            exit 1
+          fi
+          if [[ -z "$local_version" ]]; then
+            echo "failed to extract local version"
+            exit 1
+          fi
+          if [[ "$upstream_version" == "$local_version" ]]; then
+            echo "upstream ${{ matrix.components.name }} (${upstream_version}) matches local ${{ matrix.components.name }} (${local_version})"
+            exit 0
+          fi
+          echo "upstream ${{ matrix.components.name }} (${upstream_version}) is different than local ${{ matrix.components.name }} (${local_version}), creating PR"
+          git config --global user.email "actions@github.com"
+          git config --global user.name "github-actions[bot]"
+          pr_branch=ci/bump-dependencies/${{ matrix.components.name }}
+          git checkout -b "${pr_branch}"
+          # sed does not support replacements with overlapping or lookahead patterns as is the case with NSIS.
+          # Therefore, use perl instead:
+          perl -pe 's/${{ matrix.components.local_version_regex }}/${1}'"${upstream_version}"'${3}/gi' -i "${files[@]}"
+          git add .
+          title="Build: Update ${{ matrix.components.name }} to ${upstream_version}"
```

Comment:
I'd like the title to make it clear this is automation.  The body could have SKIP, really.

Reply:
I've updated the code to append `(Automated PR)` to the PR title.
Did you want to have such a comment in the commit as well? I'd rather not have it there. If the commit makes it into `master`, it will have been reviewed by a human. As such, I don't think it's worth denoting it's automated source (besides the author info, which covers it anyway).

Reply:
I think we Prefix the po file update with AUTO: on the website (not sure). We should be consistent.

Reply:
Yes, but the difference is that the .po files are [auto-committed](https://github.com/jamulussoftware/jamuluswebsite/blob/release/.github/workflows/main.yml#L41-L46) to a non-feature branch, aren't they?
In contrast, we are talking about auto-PRs here (with no intention to do that unattended).

Reply:
I'd still rather have something other than `Build` as the commit comment prefix that makes it simple to search for these changes.

Reply:
Something like [`git log --author=actions@github.com`](https://github.com/hoffie/jamulus/commit/98d2a7ea8cfdae2365b4bd4b3e11602e00e3ac75)? ;)

```
$ git log --author=actions@github.com
commit 98d2a7ea8cfdae2365b4bd4b3e11602e00e3ac75 (HEAD, hoffie/ci/bump-dependencies/aqt)
Author: github-actions[bot] <actions@github.com>
Date:   Sat Aug 13 11:18:16 2022 +0000

    Build: Update aqt to 2.2.1
```

Or do you still think we should add the AUTO: prefix here? I still think we shouldn't, but in the end it's better to add it instead of not merging this PR. ;)

Reply:
I don't have a strong opinion here. Yes, the author should make the fact that it's automated clear.

Reply:
> Something like [git log --author=actions@github.com](https://github.com/hoffie/jamulus/commit/98d2a7ea8cfdae2365b4bd4b3e11602e00e3ac75)? ;)

Hm... I guess that would look something like
```
98d2a7ea 2022-08-13 Build: Update aqt to 2.2.1 [github-actions[bot]]
340ea336 2022-08-09 (HEAD -> master, origin/master, origin/HEAD) Set version to 3.9.0dev [ann0see]
983ae3d3 2022-08-09 (tag: r3_9_0nightly1) r3.9.0Nightly1 [ann0see]
820c4aa6 2022-08-09 Merge pull request #2774 from softins/fix-new-empty-chaninfo [ann0see]
```
for my git log.  So yes, I can see it was a bot and what was done.  That'll do.